pack3d/pack3d.go

```diff
@@ -0,0 +1,356 @@
+package pack3d
```

Comment:
Cool! I've always wanted to see how the packing algorithm worked

Reply:
Yeah it's an interesting one and definitely room for improvement. If you want a deeper dive, learning about the BVH trees (representation we build for quick collision detection) and annealing (gradual improvement in packing efficiency) are great places to start.